File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/model/feedback/Feedback.java
package de.vkb.dojo.es.facilityManagement.model.feedback;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
@JsonSubTypes({
        @JsonSubTypes.Type(value = FailFeedback.class, name = "error"),
        @JsonSubTypes.Type(value = SuccessFeedback.class, name = "success")
})
public interface Feedback {
    Boolean getSuccess();
}

File: mock-human-resources/src/main/kotlin/de/vkb/dojo/es/humanResources/model/feedback/Feedback.kt
package de.vkb.dojo.es.humanResources.model.feedback

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
@JsonSubTypes(
    JsonSubTypes.Type(value = UnknownFeedback::class, name = "unknown"),
    JsonSubTypes.Type(value = FailFeedback::class, name = "error"),
    JsonSubTypes.Type(value = SuccessFeedback::class, name = "success")
)
interface Feedback {
    val success: Boolean
}
